docs/src/references/operation-workflow.md

```diff
@@ -0,0 +1,372 @@
+  - e.g. `te/device/child-xyz///cmd/configuration-update/req-123`
+- The messages published over this topic represent the current state of the command.
+  - Such a message states where the command progression is and gives all the required information to proceed.
+  - e.g. `{ "status": "Requested", "operation": "ConfigurationUpdate", "target": "mosquitto", "url": "https://..."}`
+- The state messages are published as retained.
+  - They capture the latest state of the operation request.
+  - Till some change occurs, this latest state is dispatched to any participant on reconnect.
+- Several participants act in concert to move forward the command execution.
+  - The participants observe the progress of all the operations they are interested in.
+  - They watch for the specific states they are responsible in moving forward.
+  - When a step is performed, successfully or not, the new state is published accordingly by the performer.
+
+### Example
+
+Here is an example where three software components participate in a `configuration-update` command.
+- The `tedge-mapper` creates the initial state of the command
+  providing the required information to install a new version for a configuration file;
+  and then waits for the final outcome (in black).
+- The `tedge-config-plugin` handles the main steps (in red): downloading the file and installing it where expected.
+- An extra component, provided by the user, handles domain-specific checks (in blue)
+  to timely schedule the command as well as to ensure the configuration file is not corrupted and properly installed.
+
+```mermaid
+stateDiagram-v2
+    [*] --> Init
+    Init --> Download
+    Download --> Downloaded
+    Downloaded --> Install
+    Install --> Installed
+    Installed --> Successful
+    Init --> Failed
+    Download --> Failed
+    Downloaded --> Failed
+    Install --> Failed
+    Installed --> Failed
+    Successful --> [*]
+    Failed --> [*]
+    
+    classDef specific color:blue;
+    class Init, Downloaded, Installed specific
+    
+    classDef plugin color:red;
+    class Download, Install plugin
+       
+    classDef mapper color:black;
+    class Successful, Failed mapper
+```
+
+### Benefits
+
+Observe on the example that:
+
+- At any state, *one and only one* participant is responsible to move the operation forward.
+- Publishing a state to the MQTT command topic, can be seen as passing the baton from one participant to another.
+  The mapper creates the `Init` state and let then the other components work.
+  The plugin tell the download has been successful by publishing the `Downloaded` state,
+  but do nothing till the domain-specific component has checked the file and move the command state to `Install`.
+- Each software component has to know only *some* states of the whole workflow:
+  - the states they are responsible for
+  - the states they create to pass the baton.
```

Comment:
How about using `control` instead of `baton`?

Reply:
changed